src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/InsertManyCommandResolver.java

```diff
@@ -29,6 +33,11 @@ public Class<InsertManyCommand> getCommandClass() {
 
   @Override
   public Operation resolveCommand(CommandContext ctx, InsertManyCommand command) {
+    if (ctx.embeddingService() != null) {
+      new VectorizeData(ctx.embeddingService(), objectMapper.getNodeFactory())
```

Comment:
One thought, seeing this pattern being repeated multiple times: maybe `CommandContext` could have a method like `tryVectorize(List<Document>)` since it has "embeddingService" (if any), and likely also `ObjectMapper`.
So call would be:

    ctxt.tryVectory(command.documents());

and could check existence (or lack thereof) of embedding service within method.